cypress/integration/dunning/create_dunning_candidates.js

```diff
@@ -0,0 +1,211 @@
+  const productName = 'Convenience Salat 250g';
+  const originalQuantity = 200;
+
+
+  // Test data
+  let siDocumentNumber;
+  let siCurrency;
+  let siDueDate;
+  let siTotalAmount;
+
+
+  before(function () {
+    // This wait is stupid.
+    // It also appears to be a good workaround for the problems in
+    // cypress/support/utils/utils.js:1
+    cy.wait(5000);
+  });
+
+  it('Prepare dunning type', function () {
+
+    cy.fixture('settings/dunning_type.json').then(dunningType => {
+      Object.assign(new DunningType(), dunningType)
+        .setName(dunningTypeName)
+        .apply();
+    });
+  });
+
+  it('Prepare customer bpartner (via api)', function () {
+    cy.fixture('sales/simple_customer.json').then(customerJson => {
+      const bpartner = new BPartner({...customerJson, name: businessPartnerName})
+        .setCustomer(true)
+        .setDunning(dunningTypeName)
+        .setPaymentTerm(paymentTerm)
+        .setBank(undefined);
+
+      bpartner.apply();
+    });
+  });
+
+  it('Prepare sales invoice', function () {
+    cy.fixture('sales/sales_invoice.json').then((salesInvoiceJson) => {
+      new SalesInvoice(businessPartnerName, salesInvoiceTargetDocumentType)
+        .addLine(
+          new SalesInvoiceLine().setProduct(productName).setQuantity(originalQuantity)
+          // todo @dh: how to add a "per test" packing item
+          // .setPackingItem('IFCO 6410 x 10 Stk')
+          // .setTuQuantity(2)
+        )
+        // .addLine(
+        // todo @dh: how to add this line which depends on the packing item?
+        //   new SalesInvoiceLine()
+        //     .setProduct('IFCO 6410_P001512')
+        //     .setQuantity(2)
+        // )
+        // .setPriceList(priceListName)
+        .setDocumentAction(getLanguageSpecific(salesInvoiceJson, DocumentActionKey.Complete))
+        .setDocumentStatus(getLanguageSpecific(salesInvoiceJson, DocumentStatusKey.Completed))
+        .apply();
+    });
+  });
+
+  // it('use already existing sales invoice', function () {
+  //   cy.visitWindow('167', '1000001');
+  // });
+
+
+  it('Sales Invoice is Completed', function () {
+    cy.expectDocumentStatus(DocumentStatusKey.Completed);
+  });
+
+  it('Sales Invoice is not paid', function () {
+    cy.getCheckboxValue('IsPaid').then(checkBoxValue => {
+      cy.log(`IsPaid = ${checkBoxValue}`);
+      assert.equal(checkBoxValue, false);
+    });
+  });
+
+  it('Save values needed for the next step', function () {
+    cy.getStringFieldValue('DocumentNo').then(documentNumber => {
+      siDocumentNumber = documentNumber;
+    });
+
+    cy.getStringFieldValue('C_Currency_ID').then(currency => {
+      siCurrency = currency;
+    });
+
+    cy.getStringFieldValue('DateInvoiced').then(dueDate => {
+      siDueDate = dueDate;
+    });
+
+    cy.getSalesInvoiceTotalAmount().then(amount => {
+      siTotalAmount = amount;
+    });
+  });
+
+
+  // it('Ensure there are no Dunning Candidates', function () {
```

Comment:
pls clean this up. on real-live-instances there might always be dunning candidates

Reply:
for newly created sales order there shouldn't be any dunning candidates.